src/WCSimConstructCylinder.cc

```diff
@@ -1035,12 +1037,27 @@ G4LogicalVolume* WCSimDetectorConstruction::ConstructCaps(G4int zflip)
   // barrel cells.
   // ---------------------------------------------------------
 
+  G4double borderBlackSheetRmin[3] = { WCIDRadius, WCIDRadius, WCIDRadius};
+  G4double borderBlackSheetRmax[3] = {WCIDRadius+WCBlackSheetThickness, WCIDRadius+WCBlackSheetThickness,WCIDRadius+WCBlackSheetThickness};
+  G4Polyhedra* solidWCBarrelBorderCellBlackSheet = new G4Polyhedra("WCBarrelBorderCellBlackSheet",
+                                                   -dPhi/2., // phi start
+                                                   dPhi, //total angle 
+                                                   1, //NPhi-gon
+                                                   3,
+                                                   borderAnnulusZ,
+                                                   borderBlackSheetRmin,
+                                                   borderBlackSheetRmax);
+  G4LogicalVolume* logicWCBarrelBorderCellBlackSheet =
+    new G4LogicalVolume(solidWCBarrelBorderCellBlackSheet,
+                        G4Material::GetMaterial("Blacksheet"),
+                        "WCBarrelBorderCellBlackSheet",
+                        0,0,0);
 
-   G4VPhysicalVolume* physiWCBarrelBorderCellBlackSheet =
+  G4VPhysicalVolume* physiWCBarrelBorderCellBlackSheet =
     new G4PVPlacement(0,
                       G4ThreeVector(0.,0.,0.),
-                      logicWCBarrelCellBlackSheet,
-                      "WCBarrelCellBlackSheet",
+                      logicWCBarrelBorderCellBlackSheet,
```

Comment:
The original implementation used the same blacksheet from normal ring cell (those while rectangles enclosing the PMTs), but the top and bottom rings actually has a larger height than normal ring. This makes a region with no blacksheet around the cap.
![image](https://user-images.githubusercontent.com/19830271/223888507-35d15d9f-1449-4cd8-a4c4-9a555921a5e7.png)
 
The new implementation defines a new blacksheet which covers the missing region.
![image](https://user-images.githubusercontent.com/19830271/223890892-01c8c7ca-6afb-467d-9b9b-11c2f40ca2f1.png)